### PR TITLE
[Features]Add mysql last_day function support

### PR DESCRIPTION
**Pull request by hf200012**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exprs/timestamp_functions.cpp [hunk 1]
```diff
@@ -608,6 +608,28 @@ DateTimeVal TimestampFunctions::time_round(FunctionContext* ctx, const DateTimeV
     return new_ts_val;
 }
 
+DateTimeVal TimestampFunctions::last_day(FunctionContext* ctx, const DateTimeVal& ts_val) {
+    if (ts_val.is_null) {
+        return DateTimeVal::null();
+    }
+    const DateTimeValue& ts_value = DateTimeValue::from_datetime_val(ts_val);
+    int last_days[12] = {31,28,31,30,31,30,31,31,30,31,30,31};
```
Comment (morningman) [inline be/src/exprs/timestamp_functions.cpp:616]: this array can be a static member.

File: be/src/exprs/timestamp_functions.cpp [hunk 2]
```diff
@@ -608,6 +608,28 @@ DateTimeVal TimestampFunctions::time_round(FunctionContext* ctx, const DateTimeV
...
+    int day ;
+    int year = ts_value.year();
+    int month = ts_value.month();
+    if((year % 4 == 0 && year % 100 !=0)||(year % 400 ==0 )) {
+	 day = 29 ;
+    } else {
+         day = last_days[month - 1];
```
Comment (morningman) [inline be/src/exprs/timestamp_functions.cpp:623]: ident

**Review by morningman — COMMENTED** [review]
You need to add ut for this function, on BE side.

**Comment by hf200012** [comment]
> You need to add ut for this function, on BE side.

ok

File: be/src/exprs/timestamp_functions.cpp [hunk 3]
```diff
@@ -608,6 +608,28 @@ DateTimeVal TimestampFunctions::time_round(FunctionContext* ctx, const DateTimeV
...
+DateTimeVal TimestampFunctions::last_day(FunctionContext* ctx, const DateTimeVal& ts_val) {
+    if (ts_val.is_null) {
+        return DateTimeVal::null();
+    }
+    const DateTimeValue& ts_value = DateTimeValue::from_datetime_val(ts_val);
+    int last_days[12] = {31,28,31,30,31,30,31,31,30,31,30,31};
+    int day ;
+    int year = ts_value.year();
+    int month = ts_value.month();
+    if((year % 4 == 0 && year % 100 !=0)||(year % 400 ==0 )) {
```
Comment (yangzhg) [inline be/src/exprs/timestamp_functions.cpp:620]: Additional white space

Comment (hf200012) [reply be/src/exprs/timestamp_functions.cpp:620]: solved

File: be/src/exprs/timestamp_functions.cpp (hunk 2 above)
Comment (hf200012) [reply be/src/exprs/timestamp_functions.cpp]: solved

File: be/src/exprs/timestamp_functions.cpp (hunk 1 above)
Comment (hf200012) [reply be/src/exprs/timestamp_functions.cpp]: solved

**Comment by EmmyMiao87** [comment]
It’s better to link an issue for new features ~

**Pull request closed without merging**